### DISCUSSION TITLE
Disable authentication in self-hosted deployments

### DISCUSSION THREAD

**Discussion by mariusp12** [langfuse/langfuse#4212] [Ideas]
### Describe the feature or potential improvement

For self-hosted deployments, it should be possible to disable authentication completely. Currently either a username/password authentication or a SSO is mandatory.

This could be controlled over an environment variable (for example AUTH_DISABLE)

**Comment by marcklingen** [comment]
Thanks for sharing. You can disable email/password authentication via `AUTH_DISABLE_USERNAME_PASSWORD`, but this will basically disable the whole platform. I agree that anonymous use would be interesting.

One workaround could be to [auto-provision a default user/project/org via headless initialization](https://langfuse.com/docs/deployment/self-host#initialization) and then also disable new user creation via `AUTH_DISABLE_SIGNUP`. Thereby everyone can sign in with this default user without needing to set up own accounts.

**Reply by mariusp12** [reply]
@marcklingen Thank you for the response! That was the workaroud which I used so far. However, what I would like to achieve is to have an external authentication and authorization mechanism (which cannot really be configured as langfuse custom because of several reasons). By keeping the solution you described, the user would need to introduce the password after the external login, which does not really make sense.

That is the reason why I would need to disable completely the authentication from langfuse.

**Reply by marcklingen** [reply]
Makes sense, thanks for sharing the context, let's track overall interest for this in this thread!
